### PR TITLE
Pydantiv v1/v2 compatibility fixes.

### PR DESCRIPTION
**Pull request by marius-baseten**
Convert TRT-LLM config to pydantic v1 and add integration tests for running user models with both pydantic v1 and v2.

## :rocket: What
* Downgrade TRT-LLM config to use v1.
* Add two test cases for user trusses to use pydantic v1 and v2 respectively.

## :microscope: Testing
Integration tests included in this PR.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -21,7 +21,7 @@ repos:
         # supported by your project here, or alternatively use
         # pre-commit's default_language_version, see
         # https://pre-commit.com/#top_level-default_language_version
-        language_version: python3.9
+        language_version: python3.11
```
Comment (bolasim) [inline .pre-commit-config.yaml:24]: can you revert this to 3.9 so that the precommit passes

**Review by bolasim — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (marius-baseten) [reply .pre-commit-config.yaml]: Yes, I just had to do this locally to somehow get through my pre-commits in order to push this...

File: truss/tests/test_model_inference.py [hunk 2]
```diff
@@ -187,6 +187,26 @@ def test_requirements_file_truss():
         assert response.json() is False
 
 
+@pytest.mark.parametrize("pydantic_major_version", ["1", "2"])
+def test_requirements_pydantic_v1(pydantic_major_version):
```
Comment (bolasim) [inline truss/tests/test_model_inference.py:191]: Please mark this as an integration test since it uses docker.

Comment (bolasim) [reply truss/tests/test_model_inference.py:191]: `@pytest.mark.integration`

**Pull request merged**